frontend/src/modules/frameworkImport/views/frameworkImportView.js

```diff
@@ -75,22 +75,22 @@ define(function(require){
       Origin.router.navigateToHome();
     },
 
-    onFormSubmitSuccess: function(importData, impoprtStatus, importXhr) {
+    onFormSubmitSuccess: function(data, importStatus, importXhr) {
       Origin.router.navigateToHome();
     },
 
-    onAjaxSuccess: function() {
-      this.goBack();
-    },
-
     onAjaxError: function(data, status, error) {
+      var title = data.responseJSON.title || Origin.l10n.t('app.importerrortitle');
+      var msg = data.responseJSON.body.replace(/\n/g, "<br />") || error;
```

Comment:
Query: the original code had an additional truthy check for the response before doing a `.replace()`. Is it fine that this has been removed?

Reply:
I think this is only present in the case of a JSON response (duh?), or when `json` is set as the `$.ajax` `dataType`. As we also use [`Express.send`](https://expressjs.com/en/api.html#res.send), we're not guaranteed to **send** a JSON response back, so the return type may vary.